portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,51 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<String>();
+    Query query = new Query("comment");
+    PreparedQuery results = DatastoreServiceFactory.getDatastoreService().prepare(query);
+    for (Entity entity : results.asIterable()) {
+      messages.add((String)entity.getProperty("message"));
```

Comment:
Interesting, what type are Properties? Is it this function?

http://go/app-engine-docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#getProperty-java.lang.String-

Aha, it returns an Object, so you need to cast it to the String. Got it.